packages/gatsby/src/state-machines/data-layer/actions.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  assign,
+  AnyEventObject,
+  ActionFunction,
+  AssignAction,
+  DoneInvokeEvent,
+  ActionFunctionMap,
+} from "xstate"
+import { createGraphQLRunner } from "../../bootstrap/create-graphql-runner"
+import reporter from "gatsby-cli/lib/reporter"
+import { IDataLayerContext } from "./types"
+
+const concatUnique = <T>(array1?: T[], array2?: T[]): T[] =>
```

Comment:
(I'm a fan of consistency and would like us to use `Array<T>` over `T[]` because `Map`, `Set`, etc also need to do this. But that's a different discussion to be held elsewhere)